### ISSUE TITLE
Reload DownloadMode file/retrieve from URL

### ISSUE THREAD

**Issue by bored-engineer** [gomods/athens#1669]
**Is your feature request related to a problem? Please describe.**
When using [DownloadMode](https://docs.gomods.io/configuration/download/) to enforce an allow-list of acceptable modules (ala the original [FilterFile](https://docs.gomods.io/configuration/filter/) feature) the list is loaded once when Athens starts, modifying the file requires restarting Athens before the change takes effect. 

**Describe the solution you'd like**
I would like to be able to load the HCL file a URL instead, refreshing on a regular (configured) interval, ex:
```
DownloadMode: http://<host>:<port>/something.list
DownloadModeRefresh: 60s
```

**Describe alternatives you've considered**
A simple solution to this would be to reload the file (or potentially athens entire config) when a user signal is received or on a regular time interval.

**Comment by arschles** [comment]
@bored-engineer we talked in Slack:

```
If we just implemented refresh logic for the existing file case this wouldn't really be a concern right? It becomes a concern when we add fetching from a HTTP endpoint? We could enforce TLS in the provided URL but that's a bit unnecessary in localhost configurations
```

We should force the download mode URL to use SSL, and then I think this is a fine idea. Also, this is a smaller thing but if you send a PR, make sure to have Athens clearly log when it reloads config, including a timestamp

**Comment by bored-engineer** [comment]
@arschles Looking further I think my use-case is better suited by the existing ValidationMiddleware implementation, however I don't see any documentation on it beyond the code itself, am I missing some caveat where it won't work or I should prefer to filter modules via FilterFile/DownloadMode instead?

**Comment by arschles** [comment]
@bored-engineer good idea on using the middleware. we haven't documented it at all at the moment (just filed https://github.com/gomods/athens/issues/1679 for that). Wouldn't using DownloadMode take you back to your original problem? Anyway, for now if you set the validation URL (`ATHENS_PROXY_VALIDATOR` env var), Athens ends up doing a POST request to that URL (see [here](https://github.com/gomods/athens/blob/46b68e4da8acce8198f8ffd41e405d8efd046bcb/pkg/middleware/validation.go#L79)). If you return a 200 status code from that server, it'll accept the module and otherwise it'll reject it.
